web/src/beta/features/Editor/tabs/widgets/SidePanel/ContainerSettings/hooks.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from "react";
+
+import { useWidgetsFetcher } from "@reearth/services/api";
+import { type WidgetAreaState, selectedWidgetAreaVar } from "@reearth/services/state";
+
+export default ({ sceneId }: { sceneId: string }) => {
+  const { useUpdateWidgetAlignSystem } = useWidgetsFetcher();
+  const handleAreaStateChange = useCallback(
+    async (widgetAreaState?: WidgetAreaState) => {
+      if (!sceneId || !widgetAreaState) return;
+      const results = await useUpdateWidgetAlignSystem(widgetAreaState, sceneId);
+      if (results.status === "success") {
+        selectedWidgetAreaVar(widgetAreaState);
+      }
+    },
+    [sceneId, useUpdateWidgetAlignSystem],
+  );
```

Comment:
We already import useWidgetsFetcher in SidePanel hooks and we are doing logic around widgetArea selection there, so can you move this there too to keep related logic together? 🙏🏻

Reply:
Updated!